addons/account/tests/test_payment.py

```diff
@@ -170,12 +171,159 @@ def test_payment_chf_journal_usd(self):
             'currency_id': self.currency_chf_id,
             'journal_id': self.bank_journal_usd.id,
             'partner_type': 'supplier',
-            'partner_id': self.partner_axelor.id,
+            'partner_id': self.partner_china_exp.id,
             'payment_method_id': self.payment_method_manual_out.id,
         })
         payment.post()
 
         self.check_journal_items(payment.move_line_ids, [
             {'account_id': self.account_usd.id, 'debit': 0.0, 'credit': 38.21, 'amount_currency': -58.42, 'currency_id': self.currency_usd_id},
-            {'account_id': self.partner_axelor.property_account_payable_id.id, 'debit': 38.21, 'credit': 0.0, 'amount_currency': 50, 'currency_id': self.currency_chf_id},
+            {'account_id': self.partner_china_exp.property_account_payable_id.id, 'debit': 38.21, 'credit': 0.0, 'amount_currency': 50, 'currency_id': self.currency_chf_id},
+        ])
+
+    def test_multiple_payments_00(self):
+        """ Create test to pay several vendor bills/invoices at once """
+        # One payment for inv_1 and inv_2 (same partner)
+        inv_1 = self.create_invoice(amount=100, currency_id=self.currency_eur_id, partner=self.partner_agrolait.id)
+        inv_2 = self.create_invoice(amount=500, currency_id=self.currency_eur_id, partner=self.partner_agrolait.id)
+        # One payment for inv_3 (different partner)
+        inv_3 = self.create_invoice(amount=200, currency_id=self.currency_eur_id, partner=self.partner_china_exp.id)
+        # One payment for inv_4 (Vendor Bill)
+        inv_4 = self.create_invoice(amount=50, currency_id=self.currency_eur_id, partner=self.partner_agrolait.id, type='in_invoice')
+
+        ids = [inv_1.id, inv_2.id, inv_3.id, inv_4.id]
+        register_payments = self.register_payments_model.with_context(active_ids=ids).create({
+            'payment_date': time.strftime('%Y') + '-07-15',
+            'journal_id': self.bank_journal_euro.id,
+            'payment_method_id': self.payment_method_manual_in.id,
+        })
+        register_payments.create_payment()
+        payment_ids = self.payment_model.search([('invoice_ids', 'in', ids)], order="id desc")
+
+        self.assertEqual(len(payment_ids), 3)
+        self.assertAlmostEquals(register_payments.amount, 750)
+
+        inv_1_2_pay = None
+        inv_3_pay = None
+        inv_4_pay = None
+        for payment_id in payment_ids:
+            self.assertEqual('posted', payment_id.state)
+            if payment_id.partner_id == self.partner_agrolait:
+                if payment_id.partner_type == 'supplier':
+                    self.assertEqual(payment_id.amount, 50)
+                    inv_4_pay = payment_id
+                else:
+                    self.assertEqual(payment_id.amount, 600)
+                    inv_1_2_pay = payment_id
+            else:
+                self.assertEqual(payment_id.amount, 200)
+                inv_3_pay = payment_id
+
+        for payment_id in payment_ids:
```

Comment:
Yeah... I guess it was to make sure the review was done carefully... or something like that...